### ISSUE TITLE
Why is main_pretrain.py using classification_dataloader and not pretrain_dataloader?

### ISSUE THREAD

**Issue by ChintanTrivedi** [vturrisi/solo-learn#102]
Trying to load a custom dataset without labels only for pretraining (not linear evaluation). 

I implemented the necessary changes to the pipeline in `pretrain_dataloader.py`, but I realized upon running `main_pretrain.py` (line 88) calls `classification_dataloader.py` and not `pretrain_dataloader.py` as I was expecting. Please advise how to address this for loading such a custom dataset.

**Comment by vturrisi** [comment]
We use a classification dataloader that contains the validation data in order to estimate the linear classification accuracy in an online fashion. If you just remove the `val_loader` from `trainer.fit(model, train_loader, val_loader)` no validation step will be performed. 

It's in our pipeline to allow custom datasets with/without labels in an easier way.

**Comment by ChintanTrivedi** [comment]
Thanks @vturrisi, removing `val_loader` worked for the dataloader but I'm facing the same issue since `training_step` from `base.py` line 336 is called. This method requires to compute `loss=F.cross_entropy` for classification loss in `_shared_step`, which I would think is not essential for training since it's unsupervised.

**Comment by vturrisi** [comment]
We are now allowing custom data to be used. We did some tests and it's working, but if you encounter any problems let us know. 

You can follow this example file (https://github.com/vturrisi/solo-learn/blob/main/bash_files/pretrain/custom/byol.sh). You can also set different sizes for the crops and mean and std for the normalization with --size, --mean and --std.

Let me know if something is not clear :)

**Comment by ChintanTrivedi** [comment]
Many thanks, was able to get my custom dataset without labels to work very well with simclr so far (yet to run on others). Appreicate your hard work and quick response on this :)
